### PR TITLE
Update go.mod dependencies release-0.17 [SECURITY]

### PR DESCRIPTION
**Pull request by turkenf**
### Description of your changes

This PR updates `go.mod` dependencies to fix the following:

| Name | Change | Type | Vulnerability | Severity |
|---|---|---|---|---|
| golang.org/x/crypto | `v0.21.0` -> `v0.31.0` | go-module | GHSA-v778-237x-gjrc | High |

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

https://github.com/upbound/provider-terraform/actions/runs/12396612822

### CODE REVIEW

**Comment by turkenf** [comment]
/test-examples="examples/workspace-inline-aws.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/workspace-inline-aws.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/workspace-inline.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/workspace-inline-aws.yaml"

File: examples/providerconfig-aws.yaml [hunk 1]
```diff
@@ -27,5 +27,5 @@ spec:
       }
       provider "aws" {
         shared_credentials_files = ["${path.module}/aws-creds.ini"]
-        region = "eu-west-1"
+        region = "us-east-2"
```
Comment (sergenyalcin) [inline examples/providerconfig-aws.yaml:30]: This was changed to unblock the VPC quota.

**Review by sergenyalcin — APPROVED** [review]
Thanks @turkenf LGTM!

**Pull request merged**